src/modules/decentralized-web-node/decentralized-web-node-response.interceptor.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+import { Observable, catchError } from 'rxjs';
+import { ResponseObjectDto } from './dtos/response-object.dto';
+
+@Injectable()
+export class DecentralizedWebNodeInterceptor
+  implements NestInterceptor<ResponseObjectDto>
+{
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler
+  ): Observable<ResponseObjectDto> {
+    const request: Request = context.switchToHttp().getRequest();
+    const response: Response = context.switchToHttp().getResponse();
+
+    return next.handle().pipe<ResponseObjectDto>(
+      catchError(async (err) => {
+        const status = err.status || 500;
+        console.log(err);
+        response.status(status);
+        return {
+          requestId: request.body.requestId || '',
+          status: {
+            code: status,
+            message: err.message,
```

Comment:
Spec says that the message should be "The request could not be processed correctly". What do you think of this? To me it makes sense, as the server shouldn't necessarily reveal the internal error to the client.
https://identity.foundation/decentralized-web-node/spec/#general-request-processing-error
```suggestion
            message: "The request could not be processed correctly",
```